BHoM_Engine/Compute/SHA256Hash.cs

```diff
@@ -0,0 +1,71 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2020, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Base;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Security.Cryptography;
+using System.Reflection;
+using BH.Engine.Serialiser;
+using BH.oM.Reflection.Attributes;
+using System.ComponentModel;
+
+namespace BH.Engine.Base
+{
+    public static partial class Compute
+    {
+
+        ///***************************************************/
+        ///**** Private Methods                           ****/
+        ///***************************************************/
+
+        [Description("Computes a SHA 256 hash from the given string.")]
```

Comment:
Here my reasoning to keep them separate. The methods are:
1. "Compute.Hash" takes an `IObject`, so this will obviously return an hash that is computed *"the BHoM way"*, as defined and decided by ourselves for our `IObject`. This is our "main" method, the only one that will be commonly used; hence it is an extension method.
2. SHA256 hash takes a `string`. It exposes a generic hash computation that comes from the .NET backend. It can be used to compute a SHA256 hash (and precisely that, i.e. not a SHA-512, or any other). This hash represents a string *independently* from what we define/decide in BHoM. Note that this is not an extension method.

These methods are distinctly different for me. So I think that either we hide SHA256 hash completely, or, since it is already exposed and might be useful for other purposes, we leave it separate from the other as it is now.  
Confusion should be limited as I purposely made the "main" one an extension method while the other one is not.

Reply:
As always this should be looked at from the perspective of the end user, not the developer. A hash is probably already a bit of a weird concept for the users, having two with different names just adds to the confusion. Personally, I still don't see a clear difference between the two even after your explanation.

Reply:
@adecler have a look at my latest commit.

Reply:
> Personally, I still don't see a clear difference between the two even after your explanation.

- Hash gives a hash for an `IObject` and allows for *exceptions* to be set for certain properties, fields etc., of the `IObject`. It depends on how the string is computed for the given `IObject`. It is a representation of that `IObject`.
- SHA256Hash gives a hash for a string. Just that. No exception or "customisations" are possible. If you have a string and you want a SHA256Hash for whatever need you can use this method.